File: golanglang/test/node/breadth-first/breadth-first_test.go
package breadthfirst

import (
	"reflect"
	"testing"

	node "github.com/1ChristianAlex/code-studies/internal/node"
	breadthfirst "github.com/1ChristianAlex/code-studies/internal/node/breadth-first"
)

func TestBreadthFirst(t *testing.T) {
	five := &node.ListNode[int]{Value: 5}
	four := &node.ListNode[int]{Value: 4}
	one := &node.ListNode[int]{Value: 1}
	two := &node.ListNode[int]{Value: 2}
	three := &node.ListNode[int]{Value: 3}
	zero := &node.ListNode[int]{Value: 0}

	four.Next = append(four.Next, five)

	one.Next = append(one.Next, four)
	two.Next = append(two.Next, four)
	three.Next = append(three.Next, four)

	zero.Next = append(zero.Next, one, two, three)

	if !reflect.DeepEqual(breadthfirst.NewNodeBreadth().BreadthFirst(*zero), []int{0, 1, 2, 3, 4, 5}) {
		t.Fatalf("Test fail")
	}
}

func TestBreadthFirstSecond(t *testing.T) {
	six := &node.ListNode[int]{Value: 6}
	five := &node.ListNode[int]{Value: 5}
	four := &node.ListNode[int]{Value: 4}
	two := &node.ListNode[int]{Value: 2}
	three := &node.ListNode[int]{Value: 3}
	zero := &node.ListNode[int]{Value: 0}

	four.Next = append(four.Next, five)

	two.Next = append(two.Next, four)

	zero.Next = append(zero.Next, six, two, three)

	if !reflect.DeepEqual(breadthfirst.NewNodeBreadth().BreadthFirst(*zero), []int{0, 6, 2, 3, 4, 5}) {
		t.Fatalf("Test fail")
	}
}

func TestBreadthFirstThrid(t *testing.T) {
	zero := &node.ListNode[int]{Value: 0}
	one := &node.ListNode[int]{Value: 1}
	two := &node.ListNode[int]{Value: 2}
	three := &node.ListNode[int]{Value: 3}
	four := &node.ListNode[int]{Value: 4}
	five := &node.ListNode[int]{Value: 5}
	six := &node.ListNode[int]{Value: 6}
	seven := &node.ListNode[int]{Value: 7}
	eight := &node.ListNode[int]{Value: 8}

	zero.Next = append(zero.Next, one, two, three)
	one.Next = append(one.Next, four)
	two.Next = append(two.Next, four)
	three.Next = append(three.Next, four)

	four.Next = append(four.Next, seven, five)
	five.Next = append(five.Next, seven, six)
	seven.Next = append(seven.Next, eight)

	if !reflect.DeepEqual(breadthfirst.NewNodeBreadth().BreadthFirst(*zero), []int{0, 1, 2, 3, 4, 7, 5, 8, 6}) {
		t.Fatalf("Test fail")
	}
}

File: tslang/test/nodes/breadth-first/breadth-first.test.ts
import { describe, expect, test } from "bun:test";
import { NodeBreadth } from "../../../src/nodes/breadth-first/breadth-first";
import { ListNode } from "../../../src/nodes/node-item";

describe("test breadth-first", () => {
	test("teste with node one ", () => {
		const five = new ListNode(5);
		const four = new ListNode(4);
		const one = new ListNode(1);
		const two = new ListNode(2);
		const three = new ListNode(3);
		const zero = new ListNode(0);

		four.next.push(five);

		one.next.push(four);
		two.next.push(four);
		three.next.push(four);

		zero.next.push(one, two, three);

		const instance = new NodeBreadth();

		expect(instance.breadthFirst(zero)).toEqual([0, 1, 2, 3, 4, 5]);
	});

	test("teste with node 6", () => {
		const five = new ListNode(5);
		const four = new ListNode(4);
		const six = new ListNode(6);
		const two = new ListNode(2);
		const three = new ListNode(3);
		const zero = new ListNode(0);

		four.next.push(five);

		two.next.push(four);

		zero.next.push(six, two, three);

		const instance = new NodeBreadth();

		expect(instance.breadthFirst(zero)).toEqual([0, 6, 2, 3, 4, 5]);
	});

	test("teste with node 7", () => {
		const zero = new ListNode(0);
		const one = new ListNode(1);
		const two = new ListNode(2);
		const three = new ListNode(3);
		const four = new ListNode(4);
		const five = new ListNode(5);
		const six = new ListNode(6);
		const seven = new ListNode(7);
		const eight = new ListNode(8);

		zero.next.push(one, two, three);
		one.next.push(four);
		two.next.push(four);
		three.next.push(four);

		four.next.push(seven, five);
		five.next.push(seven, six);
		seven.next.push(eight);

		const instance = new NodeBreadth();

		expect(instance.breadthFirst(zero)).toEqual([0, 1, 2, 3, 4, 7, 5, 8, 6]);
	});
});
